prog/postgres/converge_postgres_resource.rb

```diff
@@ -14,8 +14,11 @@ class Prog::Postgres::ConvergePostgresResource < Prog::Base
     hop_wait_servers_to_be_ready if postgres_resource.has_enough_fresh_servers?
 
     if postgres_resource.servers.all? { _1.vm.vm_host } || postgres_resource.location.provider == "aws"
-      # vm_host might be nil for AWS VM based PG instances
-      exclude_host_ids = (Config.development? || Config.is_e2e) ? [] : postgres_resource.servers.map { _1.vm.vm_host&.id }.compact
+      exclude_host_ids = []
+      if !(Config.development? || Config.is_e2e) && postgres_resource.location.provider == HostProvider::HETZNER_PROVIDER_NAME
+        used_data_centers = postgres_resource.servers.map { _1.vm.vm_host.data_center }.uniq
+        exclude_host_ids = VmHost.where(data_center: used_data_centers).map { _1.id }
```

Comment:
Do we have data_center information for US hosts?

Reply:
No we don't, but this code only runs for Hetzner.

In US, there is only one data center for Leaseweb, our isolation possibility there is using different racks, which we do already by placing each of our host to different rack (so we don't need to do anything for PG). I added some explanation to the commit message.